apps/tx-builder/src/components/Builder.tsx

```diff
@@ -57,6 +52,33 @@ const getInputHelper = (input: any) => {
   }
 };
 
+// Same regex used for web3@1.3.6
+const paramTypeNumber = new RegExp(/^(u?int)([0-9]*)$/);
+
+// This function is used to apply some parsing to some value types
+const parseInputValue = (input: any, value: string): string => {
+  // If there is a match with this regular expression we get an array value like the following
+  // ex: ['uint16', 'uint', '16']. If no match, null is returned
+  const isNumberInput = input.type.match(paramTypeNumber)
+
+  if (value.charAt(0) === '[') {
+    return JSON.parse(value.replace(/"/g, '"'));
```

Comment:
Parse can throw, is that OK in this case?

Reply:
To be honest I just left this code as it was. For a cleaner UX we should catch this error and return feedback to the user. If we catch this error and don't handle it on a specific way we could make the user to add a transaction with an invalid value. 

So to handle this properly would be out of scope for this ticket

Reply:
Might be worth catching it in the function that calls `parseInputValue` maybe?

Reply:
This is the output we would get in that case. Maybe is more helpful than nothing, but still not really clear:

![image](https://user-images.githubusercontent.com/25051234/124491200-487d7c80-ddb3-11eb-93ad-04683bb29f8f.png)

I added this ticket just in case
https://github.com/gnosis/safe-react-apps/issues/143

Reply:
Done :+1: